fluss-client/src/main/java/com/alibaba/fluss/client/utils/ConverterUtils.java

```diff
@@ -0,0 +1,610 @@
+                    if (value instanceof LocalTime) {
+                        LocalTime localTime = (LocalTime) value;
+                        return (int) (localTime.toNanoOfDay() / 1_000_000);
+                    } else {
+                        LOG.warn(
+                                "Field {} is not a LocalTime. Cannot convert to int millis.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
+                    if (value instanceof LocalDateTime) {
+                        return TimestampNtz.fromLocalDateTime((LocalDateTime) value);
+                    } else {
+                        LOG.warn(
+                                "Field {} is not a LocalDateTime. Cannot convert to TimestampData.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
+                    if (value instanceof Instant) {
+                        return TimestampLtz.fromInstant((Instant) value);
+                    } else if (value instanceof OffsetDateTime) {
+                        OffsetDateTime offsetDateTime = (OffsetDateTime) value;
+                        return TimestampLtz.fromInstant(offsetDateTime.toInstant());
+                    } else {
+                        LOG.warn(
+                                "Field {} is not an Instant or OffsetDateTime. Cannot convert to TimestampData.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            default:
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Unsupported type %s for field %s.",
+                                fieldType.getTypeRoot(), field.getName()));
+        }
+    }
+
+    /**
+     * Creates a field converter for converting from Row to POJO for a specific field based on its
+     * data type.
+     *
+     * @param fieldType The Fluss data type
+     * @param field The Java reflection field
+     * @return A converter for this field
+     */
+    private RowToFieldConverter createRowToFieldConverter(DataType fieldType, Field field) {
+        field.setAccessible(true);
+        Class<?> fieldClass = field.getType();
+
+        switch (fieldType.getTypeRoot()) {
+            case BOOLEAN:
+                return (row, pos) -> row.isNullAt(pos) ? null : row.getBoolean(pos);
```

Comment:
We should reuse the logic of null checks, and makes the conversion focus on not_null_object -> not_null_object conversion. You can do this by add a nullalbe wrapper on the top of a not-null wrapper.